tests/bluetooth/df/connectionless_cte_rx/src/test_set_iq_sampling_enable.c

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <stddef.h>
+#include <ztest.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <sys/byteorder.h>
+#include <host/hci_core.h>
+
+#include "common.h"
+#include "test_set_iq_sampling_enable.h"
+
+struct ut_bt_df_scan_cte_rx_params {
+	uint8_t slot_durations;
+	uint8_t cte_count;
+	uint8_t num_ant_ids;
+	uint8_t *ant_ids;
+};
+
+static uint8_t g_ant_ids[] = { 0x1, 0x2, 0x3, 0x4, 0x5 };
+
+static struct ut_bt_df_scan_cte_rx_params g_params = {
+	.slot_durations = BT_HCI_LE_ANTENNA_SWITCHING_SLOT_1US,
+	.cte_count = BT_HCI_LE_SAMPLE_CTE_ALL,
+	.num_ant_ids = ARRAY_SIZE(g_ant_ids),
+	.ant_ids = &g_ant_ids[0]
+};
+
+/* Macros delivering common values for unit tests */
+#define SYNC_HANDLE_INVALID (CONFIG_BT_PER_ADV_SYNC_MAX + 1)
+#define ANTENNA_SWITCHING_SLOT_INVALID                                         \
+	(BT_HCI_LE_ANTENNA_SWITCHING_SLOT_2US + 1)
+#define CTE_COUNT_INVALID (BT_HCI_LE_SAMPLE_CTE_COUNT_MAX + 1)
+#define SWITCH_PATTERN_LEN_INVALID                                             \
+	(CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN + 1)
+/* @brief Function sends HCI_LE_Set_Connectionless_CTE_Sampling_Enable
+ *        to controller.
+ *
+ * @param[in]sync_handle                 Handle of sync set.
+ * @param[in]sync_flags                  Flags related with sync set.
+ * @param[in]params                      CTE Sampling parameters.
+ * @param[in] enable                     Enable or disable CTE RX
+ *
+ * @return Zero if success, non-zero value in case of failure.
+ */
+int send_set_scan_cte_rx_enable(uint16_t sync_handle,
+				struct ut_bt_df_scan_cte_rx_params *params,
+				bool enable)
+{
+	struct bt_hci_cp_le_set_cl_cte_sampling_enable *cp;
+	struct net_buf *buf;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_LE_SET_CL_CTE_SAMPLING_ENABLE,
+				sizeof(*cp) + params->num_ant_ids);
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	(void)memset(cp, 0, sizeof(*cp));
+	cp->sync_handle = sys_cpu_to_le16(sync_handle);
+	cp->sampling_enable = enable ? 1 : 0;
+
+	cp->slot_durations = params->slot_durations;
+	cp->max_sampled_cte = params->cte_count;
+	if (params->num_ant_ids) {
+		if (params->ant_ids) {
+			uint8_t *dest_ant_ids =
+				net_buf_add(buf, params->num_ant_ids);
+
+			memcpy(dest_ant_ids, params->ant_ids,
```

Comment:
For some reason we much more often do `(void)memset` than `(void)memcpy`, while we should probably do latter if we do the former :D